modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-renderer/src/main/resources/META-INF/resources/js/hooks/useCreateFieldChange.es.js

```diff
@@ -0,0 +1,169 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import {debounce} from 'frontend-js-web';
+import {useMemo, useRef} from 'react';
+
+import {EVENT_TYPES} from '../actions/eventTypes.es';
+import {evaluate, mergePages} from '../util/evaluation.es';
+import {PagesVisitor} from '../util/visitors.es';
+import {usePage} from './usePage.es';
+
+let REVALIDATE_UPDATES = [];
+
+const getEditedPages = ({editingLanguageId, name, pages, value}) => {
+	const pageVisitor = new PagesVisitor(pages);
+
+	return pageVisitor.mapFields(
+		(field) => {
+			if (field.name === name) {
+				return {
+					...field,
+					localizedValue: {
+						...field.localizedValue,
+						[editingLanguageId]: value,
+					},
+					localizedValueEdited: {
+						...(field.localizedValueEdited ?? {}),
+						[editingLanguageId]: true,
+					},
+					value,
+				};
+			}
+
+			return field;
+		},
+		false,
+		true
+	);
+};
+
+const UPDATE_DELAY_MS = 50;
+
+let lastEditedPages = [];
+
+export default function useCreateFieldChange() {
```

Comment:
hey @p2kmgcl I wanted to prevent hooks from being created for this specific purpose. Just wondering if we can keep using the `useEvaluate` hook and implementing a debounce hook in conjunction with thunk. Let me try to show you an example to explain further.

```js
// Hooking nesting.

const createFieldChange = useEvaluate(useDebounce(fieldChange));

// Pseudocode to stabilize debounce

const useDebounce = (fn) => useCallback((args) => debounce(fn(args)), []);

// Save states within useRef to avoid losing reference

export const useEvaluate = (thunk) => {
	const {
		defaultLanguageId,
		editingLanguageId,
		pages,
		portletNamespace,
		rules,
	} = usePage();

	// Maybe another better name
	const evaluateProperties = useRef({
		defaultLanguageId,
		editingLanguageId,
		pages,
		portletNamespace,
		rules,
	});

	return useCallback(
		(args) =>
			thunk({
				evaluateProperties,
				...args,
			}),
		[
			evaluateProperties,
			thunk,
		]
	);
};


// Removes internal debounce
export default function fieldChange({
	evaluateProperties,

	// Or just destructure the event data
	properties,
}) {
	return (dispatch) => {
		...
	}
}
```

Well, I haven't tested that hypothesis but it can help to give you some more ideas and we can remove the use of debounce in the future and take away the centralized responsibilities within that hook.

Reply:
I had a rebase error, so I can add this now :laughing: 
I am going to try that approach (I neither liked creating that extra hook T.T), but I am afraid that the double useEvaluate(useDebounce())) wrap might cause the outdated props again. Anyway, I'll try :crossed_fingers:

Reply:
🤞🤞